### PR TITLE
poplar: Add initial support

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
@prime-zeng fyi

File: core/arch/arm/plat-poplar/main.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-poplar/main.c]: We don't the "All rights reserved" line any longer, please remove it in the other files added with Linaro copyright.

File: core/arch/arm/plat-poplar/platform_config.h [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ */
+
+#ifndef PLATFORM_CONFIG_H
+#define PLATFORM_CONFIG_H
+
+/* Make stacks aligned to data cache line length */
+#define STACK_ALIGNMENT		64
+
+/* PL011 UART */
+#define CONSOLE_UART_BASE	PL011_UART0_BASE
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-poplar/platform_config.h]: A define from `hi3798cv200.h` is used here, please include the file too.

File: core/arch/arm/plat-poplar/main.c (hunk 1 above)
Comment (jforissier) [reply core/arch/arm/plat-poplar/main.c]: @vchong FYI the new header format is documented in PR#1992
https://github.com/OP-TEE/optee_os/pull/1992/commits/453cae1374c98c2749899165715494f0f9f98a81

Comment (vchong) [reply core/arch/arm/plat-poplar/main.c]: Removed in all related files.

File: core/arch/arm/plat-poplar/platform_config.h (hunk 2 above)
Comment (vchong) [reply core/arch/arm/plat-poplar/platform_config.h]: Ah, it was in `main.c`. Will remove it there and include it here instead. Thanks for spotting!

**Comment by vchong** [comment]
All comment addressed. Please continue review. Thanks!

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by vchong** [comment]
Rebased, squashed and tag added. Ok to merge? Thanks!

**Pull request merged**